### PR TITLE
Search API Documentation

### LINKED ISSUE #320: Update Documentation with Search API Details
**Description**:
It's been a while since the documentation (i.e. `README.md`) has been updated to reflect the additions brought by the search API. We should give some technical details of the implementation as well as instructions on setting up the search API (so other facilities can use it).

**Acceptance criteria**:
- [ ] Give `README.md` a general spruce up of the entire thing
- [ ] Add details of how to setup/run the search API, and how to run DG API & search API on its own/together
- [ ] Add technical details of search API code - things like Pydantic controlling the models

### PR DESCRIPTION
**Pull request by MRichards99**
This PR will close #320, #328 

## Description
This PR updates the documentation to give details of the search API. Along the way, I've cleaned up a few existing things and removed aspects that are difficult to regularly update/get outdated rather quickly (e.g. contents, directory tree).

I've tried to include important bits of how the search API works, including details of how the mapping file works and the simple client handling (compared to DataGateway API anyway).

I've also moved the Postman collection to `util/` because the documentation implies that's where it lives, and it seems like a sensible place for it.

No version bump on this PR as there are no code changes.

## Testing Instructions
I mean, can you even test a markdown file 🤔 

- [ ] Review code
- [ ] Check GitHub Actions build
- [ ] If `icatdb Generator Script Consistency Test` CI job fails, is this because of a deliberate change made to the script to change generated data (which isn't actually a problem) or is here an underlying issue with the changes made?
- [ ] Review changes to test coverage
- [ ] Does this change mean a new patch, minor or major version should be made? If so, does one of the commit messages feature `fix:`, `feat:` or `BREAKING CHANGE:` so a release is automatically made via GitHub Actions upon merge?

## Agile Board Tracking
Connect to #328

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -854,6 +788,59 @@ The repo's collection can be easily imported into your Postman installation by o
 Postman and selecting File > Import... and choosing the Postman collection from your
 cloned DataGateway API repository.
 
+This collection has not been updated for the search API endpoints, so can only be used
+to query DataGateway API.
+
+
+
+
+# API Versioning
```
Comment (VKTB) [inline README.md:797]: It might be useful if you mention the `poetry run semantic-release print-version` command in this section.

**Review by VKTB — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -282,57 +259,11 @@ pre-commit install
...
-(major) commit type is found between the previous release and the most recent commit on
-master. When the version is bumped, a GitHub tag and release is made which contains
-the source code and the built versions of the API (sdist and wheel).
-
-
-
-
 # Running DataGateway API
+By default, the API will run on `http://localhost:5000` and all requests are made here
+e.g. `http://localhost:5000/sessions`.
```
[the remark this answers is not in the document]
Comment (MRichards99) [reply README.md:264]:
```suggestion
By default, the API will run on `http://localhost:5000` and all requests are made here
e.g. `http://localhost:5000/datagateway-api/sessions`.
```

Trying to see if I can add all these suggestions to a batch so they can be in the same commit so I'm adding some more to see how this all works

File: README.md [hunk 3]
```diff
@@ -4,47 +4,24 @@
...
-    - [ICATQuery](#icatquery)
-  - [Generating the OpenAPI Specification](#generating-the-openapi-specification)
-- [Utilities](#utilities)
-  - [Database Generator](#database-generator)
-  - [Postman Collection](#postman-collection)
-- [Updating README](#updating-readme)
+This is a Flask-based API that fetches data from an ICAT instance, and has two sets of
+endpoints, for two different use cases. The first is for
+[DataGateway](https://github.com/ral-facilities/datagateway) which has two methods of
+interfacing with an ICAT stack, using a Python-based ICAT wrapper library or using
```
[the remark this answers is not in the document]
Comment (MRichards99) [reply README.md:10]:
```suggestion
interfacing with an ICAT stack, using a
Python-based ICAT wrapper library](https://github.com/icatproject/python-icat) or using
```

File: README.md [hunk 4]
```diff
@@ -774,15 +629,93 @@ but allow for multiple session IDs to be used if required.
...
+format defined by the
+[PaNOSC data model](https://github.com/panosc-eu/search-api/blob/master/doc/data-model.md).
+To interface with ICAT, there needs to be a way of translating between this data model
+and the ICAT schema.
+
+#### Mapping between PaNOSC and ICAT Data Models
+To map between each data model, there is a JSON file (`search_api_mapping.json`) which
+defines the mappings for each PaNOSC entity (and all the attributes within them). This
+is configurable so these mappings can be changed as needed - each facility uses ICAT in
+slightly different ways; the example file shows the mappings used for ISIS which should give a good place to start.
```
[the remark this answers is not in the document]
Comment (MRichards99) [reply README.md:666]:
```suggestion
slightly different ways; the example file shows the mappings used for ISIS which should
give a good place to start.
```

File: README.md (hunk 1 above)
Comment (MRichards99) [reply README.md:799]: Resolved in d5d358c1f43573be7f0f443d4a20a9bd54a7c634

File: README.md [hunk 5]
```diff
@@ -4,47 +4,25 @@
...
-  - [Generating the OpenAPI Specification](#generating-the-openapi-specification)
-- [Utilities](#utilities)
-  - [Database Generator](#database-generator)
-  - [Postman Collection](#postman-collection)
-- [Updating README](#updating-readme)
+This is a Flask-based API that fetches data from an ICAT instance, and has two sets of
+endpoints, for two different use cases. The first is for
+[DataGateway](https://github.com/ral-facilities/datagateway) which has two methods of
+interfacing with an ICAT stack, using a
+Python-based ICAT wrapper library](https://github.com/icatproject/python-icat) or using
```
Comment (VKTB) [inline README.md:11]: This will fix it
```suggestion
[Python-based ICAT wrapper library](https://github.com/icatproject/python-icat) or using
```

**Review by VKTB — CHANGES_REQUESTED** [review]

**Review by VKTB — APPROVED** [review]

**Pull request merged**